config_src/drivers/nuopc_cap/mom_cap.F90

```diff
@@ -1746,6 +1748,16 @@ subroutine ModelAdvance(gcomp, rc)
 
         ! write restart file(s)
         call ocean_model_restart(ocean_state, restartname=restartname)
+
+        if (ESMF_AlarmIsRinging(stop_alarm, rc=rc)) then
+           write(restartname,'(A)')"ocn_stoch.res.nc"
+        else
+           write(restartname,'(A,I4.4,"-",I2.2,"-",I2.2,"-",I2.2,"-",I2.2,"-",I2.2,A)') &
+                "ocn_stoch.res.", year, month, day, hour, minute, seconds,".nc"
+        endif
+        call ESMF_LogWrite("MOM_cap: Writing stoch restart :  "//trim(restartname), &
+                           ESMF_LOGMSG_INFO)
+        call write_stoch_restart_ocn('RESTART/'//trim(restartname))
```

Comment:
This line currently doesn't affect us because write_stoch_restart_ocn subroutine appears to be empty, but shouldn't we place this call within an if statement checking whether stochastic physics is used?

Reply:
Is it OK to have get_param calls in mom_cap.F90?  Since I didn't see any uses of it here, so I wasn't sure if I should add the logic here.

Reply:
It's preferable to keep the get_param calls in subroutine [ocean_model_init](https://github.com/mom-ocean/MOM6/blob/main/config_src/drivers/nuopc_cap/mom_ocean_model_nuopc.F90#L228).

Reply:
@gustavo-marques understood,  I will push the updated code shortly.

Reply:
just merged Phil's modification